content/articles/getting-started-with-optimization-in-matlab/index.md

```diff
@@ -0,0 +1,196 @@
+
+ymin = mysimplefunc(xmin);
+```
+When using the `fmibnd` function, we give it the inputs, `mysimplefunc` and the minimum and maximum values of `x` in that order. After obtaining the minimum point in the x-axis, we can get the minimum `y` by referring to the `x`'s minimum value. We then display the result, or we can check them in the workspace.
+```Matlab
+disp(xmin)
+disp(ymin)
+```
+We obtain a similar result when we execute the code above since we used the same function as before.
+
+### Example 2
+Now let's introduce a more complex function and find its lowest point. In this example, we have a function $ f(x)=x^3-4x $ to use. We will first plot this function and then find its lowest point of the function. Just as before, we will use the `fminbnd` function to find these values. Just as before, we will create a script function `mysimplefunc.m` to define our mathematical function and then call our script to define that minimum point. The script function code is as follows;
+```Matlab
+function f = mysimplefunc(x)
+
+f = x.^3 - 4*x;
+```
+We will call this script function which contains our mathematical function in our script, plot the mathematical function and obtain the minimum point as follows;
+```Matlab
+x = -3:0.1:3;
+
+f = mysimplefunc(x);
+
+plot(x,f)
+
+xmin = fminbnd(@mysimplefunc, -3, 3);
+
+ymin = mysimplefunc(xmin);
+
+disp(xmin)
+disp(ymin)
+```
+When we run our code, the result is as follows;
+![plot of the function](/engineering-education/getting-started-with-optimization-in-matlab/opt3.png)
+![minimum point](/engineering-education/getting-started-with-optimization-in-matlab/opt4.png)
+
+### Rosenbrock's banana function
+It is a function that is a famous test case for optimization software. Given a function $f(x,y) = (1-x)^2 + 100(y-x^2)^2$, this function is known as Rosenbrock's banana function. We will first plot this function and then find its minimum point. In this case, we have two variables in our function, i.e., `x` and `y`, we cannot use the previous `fminbnd` function, but instead, we will use the `fminsearch` function to obtain this.
+> Note that the `fminbnd` function can only handle one variable while `fminsearch` can take more than one variable. 
+```Matlab
+clear
+clc
+
+[x,y] = meshgrid(-2:0.1:2, -1:0.1:3);
+
+f = (1-x).^2 + 100.*(y-x.^2).^2;  % our defined function
+
+figure(1)
+surf(x,y,f) %use surf function to plot the surf of the function
+
+figure(2)
+mesh(x,y,f)   %use mesh function to plot the surf of the function
+
+figure(3)
+surf(x,y,f)    %use surf function to plot the surf of the function, but in this case, we color the plotting
+shading interp;
+colormap (hot);     %used to color the plot
+```
+![plot of the banana function](/engineering-education/getting-started-with-optimization-in-matlab/opt5.png)
+
+We will use the `fminsearch` function to find the minimum of our given mathematical function. We will first create a function script in which we will define our function as shown below;
```

Comment:
We will is repeated in both the sentences. This makes the article sound boring even though this is a very interesting one.